docs/cartesian-product.adoc

```diff
@@ -0,0 +1,150 @@
+:page-title: Testing all test parameter combinations (Cartesian product)
+:page-description: Extends JUnit Jupiter with `@CartesianProductTest`, a special `@ParameterizedTest` which tests all combinations of its input
+
+Sometimes you want to test not only for specific inputs but also combinations of those inputs.
+`@CartesianProductTest` combines all test inputs and runs a test for each combination.
+
+[#_basic_use]
+== Basic Use
+
+`@CartesianProductTest` is used _instead_ of `@Test` or other such annotations (e.g. `@RepeatedTest`).
+The annotation can have a value (see <<Supplying a value to CartesianProductTest>>), or the test class can have a static factory method (see <<Writing a static factory method for the parameters>>) providing the arguments.
+
+== Supplying a value to CartesianProductTest
+
+If all your test parameters are strings, you can supply all input parameters simultaneously by giving a value to `@CartesianProductTest`.
+This value is the input for all parameters, and the test will try every combination it can have.
+
+```java
+@CartesianProductTest({"0", "1"})
+void threeBits(String a, String b, String c) {
+    int value = Integer.parseUnsignedInt(a + b + c, 2);
+    Assertions.assertThat(value).isBetween(0b000, 0b111);
+}
+```
+
+The test `threeBits` is executed exactly eight times, because all three input parameters can have the value "0" or "1".
+`@CartesianProductTest` tests for all input combinations; that's `2 × 2 × 2`, so eight tests in total.
+
+To demonstrate with a table:
+|===
+| # of test | value of `a` | value of `b` | value of `c`
+| 1st test  | 0            | 0            | 0
+| 2nd test  | 0            | 0            | 1
+| 3rd test  | 0            | 1            | 0
+| 4th test  | 0            | 1            | 1
+| 5th test  | 1            | 0            | 0
+| 6th test  | 1            | 0            | 1
+| 7th test  | 1            | 1            | 0
+| 8th test  | 1            | 1            | 1
+|===
+
+== Writing a static factory method for the parameters
+
+`@CartesianProductTest` can have a static factory method supplying the test parameters.
```

Comment:
Would add an introduction sentence like "If you don't only supply String values to your test method like in the example above..."